yaml_test.go

```diff
@@ -421,3 +425,71 @@ foo: baz
 		t.Error("expected YAMLtoJSONStrict to fail on duplicate field names")
 	}
 }
+
+func TestJSONObjectToYAMLObject(t *testing.T) {
+	intOrInt64 := func(i64 int64) interface{} {
+		if i := int(i64); i64 == int64(i) {
+			return i
+		}
+		return i64
+	}
+
+	tests := []struct {
+		name     string
+		input    map[string]interface{}
+		expected yaml.MapSlice
+	}{
+		{"nil", nil, nil},
+		{"empty", map[string]interface{}{}, yaml.MapSlice{}},
+		{
+			"values",
+			map[string]interface{}{
+				"bool":         true,
+				"float64":      float64(42.1),
+				"fractionless": float64(42),
+				"int":          int(42),
+				"int64":        int64(42),
+				"int64 big":    float64(math.Pow(2, 62)),
+				"map":          map[string]interface{}{"foo": "bar"},
+				"slice":        []interface{}{"foo", "bar"},
+				"string":       string("foo"),
+				"uint64 big":   float64(math.Pow(2, 63)),
+			},
+			yaml.MapSlice{
+				{"bool", true},
+				{"float64", float64(42.1)},
+				{"fractionless", int(42)},
+				{"int", int(42)},
+				{"int64", int(42)},
+				{"int64 big", intOrInt64(int64(1) << 62)},
+				{"map", yaml.MapSlice{{"foo", "bar"}}},
+				{"slice", []interface{}{"foo", "bar"}},
+				{"string", string("foo")},
+				{"uint64 big", uint64(1) << 63},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := JSONObjectToYAMLObject(tt.input)
```

Comment:
should this also produce the same result as marshaling the json object to json bytes, then calling `yaml.Unmarshal`? if so, can we do that and compare as well?

Reply:
Thank you Jordan, I was going to say that :-)

Reply:
added tests and fixed some corner cases to behave the same.